inference-engine/src/multi_device/multi_device_exec_network.hpp

```diff
@@ -67,14 +67,15 @@ class ThreadSafeBoundedQueue {
     ThreadSafeBoundedQueue() = default;
     bool try_push(T value) {
         std::lock_guard<std::mutex> lock(_mutex);
-        if (_capacity) {
+        if (_capacity > _queue.size()) {
             _queue.push(std::move(value));
+            return true;
         }
-        return _capacity;
+        return false;
     }
     bool try_pop(T& value) {
         std::lock_guard<std::mutex> lock(_mutex);
-        if (_capacity && !_queue.empty()) {
+        if (!_queue.empty()) {
```

Comment:
`_capacity == 0` is used to disable tasks scheduling. For example, when it is set in destructor other asynchronous tasks could not use that queue to return or acquire workers. And self-looped scheduling of infer tasks gracefully stop.

Reply:
thanks for the explaination, I got it, will revert this code.